Add snode address change OMQ subscription

jagerman opened the pull request:
Currently there are bugs in storage server around noticing new IP addresses.  This adds a mechanism into oxend so that storage server can subscribe to snode IP/port address changes to get them pushed as soon as oxend received the uptime proof with the new address, so that storage server will be able to update the record it uses to reach other storage servers as soon as possible.

This works by sending a "sub.snode_addr" request to oxend (and repeated occasionally to prevent the subscription from timing out), after which oxend will fire "notify.snode_addr" commands back along the connection whenever a proof is received that changes IP/port/pubkey info.

The change to the utils/ script here is for testing this:

```
$ ./lmq-watch.py ipc:///home/jagerman/.oxen/stagenet4/lokid.sock SNODEADDR
Connecting to ipc:///home/jagerman/.oxen/stagenet4/lokid.sock
Subscribed to snode addr changes
Snode address change for 13ef07a5dbc9ed07aa5720628a45d615eef0adba513210566a083964dea00502:
  144.91.127.110, :10036 (quorumnet), :0 (SS HTTPS), :0 (SS QUIC)
Snode address change for 001013496a4353101fc1438b6e21a43576a2f1bb6ca6b5861ed03c52534e0255:
  80.246.144.221, :10034 (quorumnet), :0 (SS HTTPS), :0 (SS QUIC)
Snode address change for ceda33e5a8f002e3fd341d2bf7fc05d796435664830d68ab61c5175137628164:
  45.207.195.210, :11025 (quorumnet), :0 (SS HTTPS), :0 (SS QUIC)
...
```
Note that you are unlikely to see much output from this unless your node hasn't been synced in a while, since it only reports on changed values; this particular stagenet oxend had not synced in 3-4 weeks.  A mainnet node that was fully synced received just one in my testing, for [this IP change](https://oxen.observer/tx/8373a7b0c0ce67dea33fa99ffe5ffc946f1e5af8fdf7d2f5dfbf10d1555ddbcc).

jagerman committed:
Add snode address change OMQ subscription

Currently there are bugs in storage server around noticing new IP
addresses.  This adds a mechanism into oxend so that storage server can
subscribe to snode IP/port address changes to get them pushed as soon as
oxend received the uptime proof with the new address, so that storage
server will be able to update the record it uses to reach other storage
servers as soon as possible.

Doy-lee commented on src/rpc/omq_server.cpp:
```diff
@@ -637,6 +700,31 @@ void omq_rpc::on_mempool_sub_request(oxenmq::Message& m) {
     }
 }
 
+template <typename SubType, typename... SubArgs>
```
`SubArgs` can be removed

jagerman replied:
Gone.

jagerman committed:
Make version changes trigger a contact update

The upcoming version of storage server tracks storage server versions so
that it can act differently depending on the remote's version.

This adds versions (oxend, lokinet, storage server) and triggering on
version changes to the address notifications.
